### DISCUSSION TITLE
PlotPDF resolution

### DISCUSSION THREAD

**Discussion by ankushs0128** [GreenleafLab/ArchR#660] [Questions / Documentation] [answered]
I would like to generate a high-resolution figure. Using the Plot PDF function, The zoomed-in figures especially UMAP are pixelated. Screenshot enclosed 

```
plot(p2, name = "_1_Plot-UMAP1-Clusters.pdf", ArchRProj = Differentiation_P_1, addDOC = TRUE, width = 
20, height = 20,plotList = NULL)
```

How can I improve the figure plotting resolution in ARCHR for all figures?
Thanks 
<img width="416" alt="Screenshot 2021-03-28 at 16 30 27" src="https://user-images.githubusercontent.com/32255128/112756041-9714c100-8fe3-11eb-933a-db075bc69408.png">

**Comment by rcorces** [comment, marked as answer]
Please note that I have converted this issue into a discussion topic because issue reports are only meant for problems with the software. All questions and feature requests should be reported in the discussions area. This is stated in the issue request screen and in the issue template.

![image](https://user-images.githubusercontent.com/10343046/112761277-6a46b680-8faf-11eb-8c8a-0e91b8f659f6.png)

To answer your question, `plotPDF()` is only meant to take already generated plots and print them onto a PDF. For the most part, `plotPDF()` has nothing to do with the resolution.

In your particular example, I'm assuming that this is an embedding plot that was created using `plotEmbedding()`. If you look at the parameters for `plotEmbedding()` you will see `rastr` as well as `...`. When `rastr = TRUE` you convert vector graphics into rasterized graphics and this results in the blurred dots that you mention. If you use `rastr = FALSE` each point will be stored as an individual vector at absolute resolution. However, editing files with tens of thousands of individual vectorize points is challenging for most graphics cards. So you can set `rastr = FALSE` and have perfect resolution. 

Alternatively, you can look into the `...` which allows you to pass additional parameters to `ArchR::ggPoint()` or also directly to `ggPlot`

**Reply by ankushs0128** [reply]
Apologies and Thanks

**Comment by sneddonucsf** [comment]
Hi @rcorces to piggy back off of this, how can we plot a vectorized plot for the "plotMarkers" function? I see that it does not have a "rastr = FALSE" option, nor the "..."

**Reply by rcorces** [reply]
@sneddonucsf - there was not previously an option to return this plot without rasterization. I believe this is because the number of points is very large and this could be problematic for certain environments. I've added the option via https://github.com/GreenleafLab/ArchR/commit/ee6bc0b89aeb9a5afdd971f2d8ede3c3827dd564 on the `dev_plotMarkers` branch which comes off of `release_1.0.2`. The default is still to rasterize for safety.

You can test this out by installing this new branch, restarting your R session, and reloading ArchR
```
devtools::install_github("GreenleafLab/ArchR", ref="dev_plotMarkers", repos = BiocManager::repositories())
```

Please let me know if this works for you as I have not tested this.

**Reply by sneddonucsf** [reply]
Seems to have worked, thank you!
